mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/SynchronousMvRxStateStore.kt

```diff
@@ -0,0 +1,53 @@
+package com.airbnb.mvrx.mocking
+
+import androidx.annotation.RestrictTo
+import com.airbnb.mvrx.MvRxStateStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flow
+
+/**
+ * This acts as a functional state store, but all updates happen synchronously.
+ * The intention of this is to allow state changes in tests to be tracked
+ * synchronously.
+ */
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class SynchronousMvRxStateStore<S : Any>(initialState: S, val coroutineScope: CoroutineScope) : MvRxStateStore<S> {
+
+    private val stateChannel = BroadcastChannel<S>(capacity = Channel.BUFFERED)
```

Comment:
@gpeal i like the idea, but it broke tests because it does conflate (and there are tests specifically checking against conflation). also stateflow is not cancellable, which we have checks for.

I think keeping this implementation will provide the closest implement to the original state store in a synchronous way